internal/sbi/producer/pdu_session.go

```diff
@@ -314,6 +323,20 @@ func HandlePDUSessionSMContextUpdate(smContextRef string, body models.UpdateSmCo
 				}
 			}
 
+			if smContext.UeCmRegistered {
+				problemDetails, err := consumer.UeCmDeregistration(smContext)
+				if problemDetails != nil {
+					if problemDetails.Cause != "CONTEXT_NOT_FOUND" {
+						logger.PduSessLog.Errorf("UECM_DeRegistration Failed Problem[%+v]", problemDetails)
+					}
+				} else if err != nil {
+					logger.PduSessLog.Errorf("UECM_DeRegistration Error[%+v]", err)
+				} else {
+					logger.PduSessLog.Traceln("UECM_DeRegistration successful")
+				}
+				smContext.UeCmRegistered = false
```

Comment:
should change the state in `UeCmDeregistration()`

Reply:
Thanks. Removed the code duplication and set the state inside UeCmDeregistration()